### PR TITLE
query api prototype

### PR DESCRIPTION
**Pull request by BigYu**

### CODE REVIEW

File: mt/data/handlers/query.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  const dimensionsParsed = parseDimensions(dimensions);
+  const queryString = qs.stringify({
+    metrics,
+    start: dimensionsParsed.start,
+    end: dimensionsParsed.end,
+    tagset: JSON.stringify(dimensionsParsed.tagset)
+  });
+  const uri = `http://${botanaApiDomain}/${botanaApiPath}?${queryString}&fields=[]`;
+
+  const dimensionsToCollapse = _.map(_.filter(dimensions, d => d[0] === 'eq'), d => d[1]);
```
Comment (haipeng-wang) [inline mt/data/handlers/query.js]: and for operator 'in' with only one value

Comment (BigYu) [reply mt/data/handlers/query.js]: Let's discuss the condition to collapse later. Keep it simple in this prototype

File: mt/data/handlers/query.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
+const rp = require('request-promise');
+const _ = require('lodash');
+const qs = require('query-string');
+
+function parseDimensions(dimensions) {
+  return _.reduce(dimensions, (memo, dimension) => {
+    const [operator, dimensionId, ...dimensionValues] = dimension;
+
+    if (dimensionId === 'time') {
```
[the remark this answers is not in the document]
Comment (haipeng-wang) [reply mt/data/handlers/query.js]: [start, end=start] = dimensionValues

[the remark this answers is not in the document]
Comment (haipeng-wang) [reply mt/data/handlers/query.js]: do it from UI side

**Pull request merged**